src/anndata/_core/anndata.py

```diff
@@ -561,125 +537,18 @@ def shape(self) -> tuple[int, int]:
     @property
     def X(self) -> XDataType | None:
         """Data matrix of shape :attr:`n_obs` × :attr:`n_vars`."""
-        if self.isbacked:
-            if not self.file.is_open:
-                self.file.open()
-            X = self.file["X"]
-            if isinstance(X, h5py.Group):
-                X = sparse_dataset(X)
-            # This is so that we can index into a backed dense dataset with
-            # indices that aren’t strictly increasing
-            if self.is_view:
-                X = _subset(X, (self._oidx, self._vidx))
-        elif self.is_view and self._adata_ref.X is None:
-            X = None
-        elif self.is_view:
-            X = as_view(
-                _subset(self._adata_ref.X, (self._oidx, self._vidx)),
-                ElementRef(self, "X"),
-            )
-        else:
-            X = self._X
-        return X
-        # if self.n_obs == 1 and self.n_vars == 1:
-        #     return X[0, 0]
-        # elif self.n_obs == 1 or self.n_vars == 1:
-        #     if issparse(X): X = X.toarray()
-        #     return X.flatten()
-        # else:
-        #     return X
+        return self.layers.get(None)
 
     @X.setter
-    def X(self, value: XDataType | None):  # noqa: PLR0912
-        if value is None:
```

Comment:
I think all this deleted logic contains things that `layers` can't currently do.  For example, overriding the backing `X`; currently, if you try to override a part of a view of a layers, it will just actualize and not override the old data.  I'm not sure what the history is there, but it accounts for a pretty big discrepancy in functionality